### ISSUE TITLE
Mini-migrator to add `build: error_overlinking: true`

### ISSUE THREAD

**Issue by isuruf** [regro/cf-scripts#1383]
This should catch some errors like wrong blas linking.

**Comment by beckermr** [comment]
Some feedstocks rely on this not being enforced. I'm not in favor of adding this by default.

**Comment by isuruf** [comment]
I think there are only a few, and people could easily turn it off by adding `false` there and the bot would ignore those feedstocks.

**Comment by beckermr** [comment]
Do you mean the bot or the mini migrator?

Also, if you really want this on by default, why not add it to smithy and an opt-out in the conda forge config?

**Comment by beckermr** [comment]
I'd very much like to see statistics on how many feedstocks rely on over linking before we turn this on.

**Comment by isuruf** [comment]
I mean the mini-migrator.

> Also, if you really want this on by default, why not add it to smithy and an opt-out in the conda forge config?

I was going to, but then builds will start failing suddenly and maintainers will not know what caused it.

> I'd very much like to see statistics on how many feedstocks rely on over linking before we turn this on.

overlinking name here is a misnomer. The issue here is under specifying dependencies, and I'd bet there'd be a handful of packages that should be underspecifying dependencies. (gsl comes to mind, but that's it)

**Comment by beckermr** [comment]
Right. My worry with the mini migrator is the same as smithy. If we just add it, bot prs will fail suddenly. 

We should do this like we are doing matplotlib to matplotlib-base. We can send a bot pr suggesting this change. We let that go for a while and then make an announcement that the default will change. Then a while after that we switch the default in smithy.
